lib/user_data_parser/row/row_parser.rb

```diff
@@ -0,0 +1,30 @@
+require_relative "./row.rb"
+
+module UserDataParser
+  class RowParser
+    attr_reader :delimiters, :input_type
+
+    def initialize(delimiters, input_type)
+      @delimiters = delimiters
+      @input_type = input_type
+    end
+
+    def parse(data)
+      values = data.split(delimiters[input_type[:delimiter]])
+      parsed_row_data = {}
+      values.each_with_index do |v, i|
+        parsed_row_data[input_type[:column_order][i]] = v.strip
```

Comment:
I think this calls out the vague intent of ```input_types``` that I brought up earlier. One thing that might mitigate it at least in this usage is to implement an explainer variable like
```column_name = input_type[:column_order][i]```